### PR TITLE
Snap 3005 new

### PR DESCRIPTION
**Pull request by BhattChandresh**
## Changes proposed in this pull request
- Create the table in external Hive Meta Store, run the queries on it from Snappy, dropping hive tables from Snappy.
- Create Hive schema from Snappy, create Hive table in Hive schema from Snappy, load the data to Hive table from Snappy, drop the Hive table and the Hive schema from Snappy.
- Create Hive table, insert data to Hive table, alter the Hive table name from Snappy, drop the Hive table from Snappy.
- Execute the join queries between the Hive tables and Snappy Tables.

## Patch testing

None

## ReleaseNotes.txt changes

None

## Other PRs 

None

### CODE REVIEW

File: dtests/src/test/java/io/snappydata/hydra/externalHiveMetaStore/hiveMetaStore.java [hunk 1]
```diff
@@ -1,4 +1,418 @@
...
+
+    private static String createHiveTables(String tableDefinition) {
+        return "create table " + tableDefinition + " row format delimited fields terminated by ',' ";   //tblproperties(\"skip.header.line.count\"=\"1\")"
+    }
+
+    private static String loadDataToHiveTbls(String dataPath, String tblName) {
+        return "load data local inpath '" + dataPath + "' overwrite into table " + tblName;
+    }
+
+    private static void validationRoutine(String dir, ResultSet rs1, ResultSet rs2, String index) {
```
Comment (sonalsagarwal) [inline dtests/src/test/java/io/snappydata/hydra/externalHiveMetaStore/hiveMetaStore.java]: Give an appropriate name to this method.

File: dtests/src/test/java/io/snappydata/hydra/externalHiveMetaStore/hiveMetaStore.java [hunk 2]
```diff
@@ -1,4 +1,418 @@
...
+            //"SELECT o.OrderID, c.CompanyName, e.FirstName, e.LastName FROM snappy_orders o JOIN snappy_employees e ON (e.EmployeeID = o.EmployeeID) JOIN snappy_customers c ON (c.CustomerID = o.CustomerID) WHERE o.ShippedDate > o.RequiredDate AND o.OrderDate > Cast('1998-01-01' as TIMESTAMP) ORDER BY c.CompanyName"
+    };
+
+    static hiveMetaStore metaStore = new hiveMetaStore();
+
+    public hiveMetaStore() {
+    }
+
+    // Presently not in use - Experimental.
+    public static void HydraTask_Wait() {
```
[the remark this answers is not in the document]
Comment (BhattChandresh) [reply dtests/src/test/java/io/snappydata/hydra/externalHiveMetaStore/hiveMetaStore.java]: Done.

File: dtests/src/test/java/io/snappydata/hydra/externalHiveMetaStore/hiveMetaStore.java (hunk 1 above)
Comment (BhattChandresh) [reply dtests/src/test/java/io/snappydata/hydra/externalHiveMetaStore/hiveMetaStore.java]: Done

File: dtests/src/test/java/io/snappydata/hydra/externalHiveMetaStore/hiveMetaStore.java [hunk 3]
```diff
@@ -1,4 +1,418 @@
...
+    public static void HydraTask_ConnectToSnappy() {
+        try {
+            metaStore.snappyConnection = SnappyTest.getLocatorConnection();
+            Log.getLogWriter().info("Snappy : Connection is successful");
+        }catch (SQLException se) {
+            throw new TestException("Snappy : Exception in  acquiring connection", se);
+        }
+    }
+
+    public static void HydraTask_DropAllTheTblsIfPresent() {
```
[the remark this answers is not in the document]
Comment (BhattChandresh) [reply dtests/src/test/java/io/snappydata/hydra/externalHiveMetaStore/hiveMetaStore.java]: Done

File: dtests/src/test/java/io/snappydata/hydra/externalHiveMetaStore/externalHiveMetaStore.conf [hunk 4]
```diff
@@ -1,14 +1,53 @@
...
 INCLUDE $JTESTS/io/snappydata/hydra/northwind/startDualModeCluster.conf;
 
-TASK       taskClass   = io.snappydata.hydra.cluster.SnappyTest taskMethod = HydraTask_executeSnappyJob
-           io.snappydata.hydra.cluster.SnappyPrms-jobClassNames = io.snappydata.hydra.complexdatatypes.ArrayType
-           io.snappydata.hydra.cluster.SnappyPrms-userAppName = "ComplexType_ArrayType_Validation"
-           io.snappydata.hydra.cluster.SnappyPrms-userAppJar = snappydata-store-scala-tests*tests.jar
-           threadGroups = snappyThreads
-           maxTimesToRun = 1;
+INITTASK taskClass = io.snappydata.hydra.externalHiveMetaStore.hiveMetaStore taskMethod = HydraTask_ConnectToBeeline
+threadGroups = snappyThreads;
```
[the remark this answers is not in the document]
Comment (BhattChandresh) [reply dtests/src/test/java/io/snappydata/hydra/externalHiveMetaStore/externalHiveMetaStore.conf]: Done.

File: dtests/src/test/java/io/snappydata/hydra/externalHiveMetaStore/externalHiveMetaStore.conf [hunk 5]
```diff
@@ -1,14 +1,53 @@
...
+INITTASK taskClass = io.snappydata.hydra.externalHiveMetaStore.hiveMetaStore taskMethod = HydraTask_JoinBetweenHiveAndSnappy
+threadGroups = snappyThreads;
+
+CLOSETASK taskClass = io.snappydata.hydra.externalHiveMetaStore.hiveMetaStore taskMethod = HydraTask_DropBeelineTablesFromSnappy
+threadGroups = snappyThreads;
+
+CLOSETASK taskClass = io.snappydata.hydra.externalHiveMetaStore.hiveMetaStore taskMethod = HydraTask_DropSnappyTables
+threadGroups = snappyThreads;
+
+CLOSETASK taskClass = io.snappydata.hydra.externalHiveMetaStore.hiveMetaStore taskMethod = HydraTask_DisconnectBeeline
```
[the remark this answers is not in the document]
Comment (BhattChandresh) [reply dtests/src/test/java/io/snappydata/hydra/externalHiveMetaStore/externalHiveMetaStore.conf]: Done.

File: dtests/src/test/java/io/snappydata/hydra/externalHiveMetaStore/hiveMetaStore.java [hunk 6]
```diff
@@ -1,4 +1,418 @@
...
+    Statement snappyST = null;
+    ResultSetMetaData rsmd = null;
+
+    static String setexternalHiveCatalog = "set spark.sql.catalogImplementation=hive";
+    static String setexternalInBuiltCatalog = "set spark.sql.catalogImplementation=in-memory";
+    static String showTblsApp = "show tables in app";
+    static String showTblsDefault = "show tables in default";
+    static String dropTableStmt = "drop table if exists ";
+    static String createDB = "create database ";
+    static String dataPath = "/home/cbhatt/NW_1GB/NW_1GB/";
```
[the remark this answers is not in the document]
Comment (BhattChandresh) [reply dtests/src/test/java/io/snappydata/hydra/externalHiveMetaStore/hiveMetaStore.java]: Done

**Pull request merged**